packages/react-ui-kit/src/Form/SelectComponents.tsx

```diff
@@ -134,6 +135,55 @@ export const Menu = (dataUieName: string, css?: CSSObject) => (props: MenuProps)
   );
 };
 
+// eslint-disable-next-line react/display-name
+export const MenuList = (menuListHeading: string, dataUieName: string) => (props: MenuListProps) => {
+  const {selectProps, children} = props;
+
+  const handleClose = () => {
+    if (selectProps && selectProps.onMenuClose) {
+      selectProps.onMenuClose();
+    }
+  };
+
+  return (
+    <components.MenuList {...props}>
+      <div
+        {...(dataUieName && {
+          'data-uie-name': `menu-list-${dataUieName}`,
+        })}
+      >
+        <div
+          css={(theme: Theme) => ({
+            display: 'flex',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+            fontSize: theme.fontSizes.medium,
+            fontWeight: 600,
+            padding: '8px 16px',
+          })}
+        >
+          {menuListHeading}
+          <button
+            onClick={handleClose}
+            css={{
+              background: 'transparent',
+              border: 'none',
+              cursor: 'pointer',
+              padding: 0,
+              display: 'flex',
+              alignItems: 'center',
+            }}
```

Comment:
I would create a new file for these styles. I see it's not common in this repo, but if you take a look at the app repository, I'd see the pattern with `<Component>.styles.ts`